runtime/src/iree/hal/drivers/cuda/tracing.c

```diff
@@ -16,8 +16,33 @@
 // To prevent spilling pages we leave some room for the context structure.
 #define IREE_HAL_CUDA_TRACING_DEFAULT_QUERY_CAPACITY (16 * 1024 - 256)
 
+// iree_hal_cuda_tracing_context_event_t contains a hipEvent that is used to
+// record timestamps. This event also belongs to 2 linked list.
```

Comment:
".. record timestamps for tracing GPU execution. In this struct, there are also two linked list that the current event may be added to during its lifetime."